### PR TITLE
Fixes connected to uniter.TestUniterDyingReaction

### PR DESCRIPTION
**Pull request by alesstimec**
(Review request: http://reviews.vapour.ws/r/2489/)

### CODE REVIEW

**Comment by axw** [comment]
Reviewing here because RB isn't showing the diff properly.

File: worker/uniter/leadership/resolver.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func (l *leadershipResolver) NextOp(
 ) (operation.Operation, error) {
 
 	// TODO(wallyworld) - maybe this can occur before install
-	if !opState.Installed {
+	if !opState.Installed || opState.Kind == operation.RunHook {
```
Comment (axw) [inline worker/uniter/leadership/resolver.go]: You're right that we shouldn't do *most* of the leadership stuff during hook error, but there's a bit in ModeHookError that deals with resigning leadership (only)

File: worker/uniter/resolver.go [hunk 2]
```diff
@@ -32,6 +32,10 @@ func (s *uniterResolver) NextOp(
 	remoteState remotestate.Snapshot,
 ) (operation.Operation, error) {
 
+	if remoteState.Life == params.Dead {
```
Comment (axw) [inline worker/uniter/resolver.go]: You'll want to rebase, I've made a slightly different change in parallel.

Comment (alesstimec) [inline worker/uniter/resolver.go]: done.

**Comment by alesstimec** [comment]
$$merge$$

**Pull request merged**